lib/internal/http2/core.js

```diff
@@ -819,6 +832,24 @@ class Http2Session extends EventEmitter {
     debug(`Http2Session ${sessionName(type)}: created`);
   }
 
+  // Returns undefined if the socket is not yet connected, true if the
+  // socket is a TLSSocket, and false if it is not.
+  get encrypted() {
+    return this[kEncrypted];
+  }
+
+  // Returns undefined if the socket is not yet connected, `h2` if the
+  // socket is a TLSSocket and the alpnProtocol is `h2`, or `h2c` if the
+  // socket is not a TLSSocket.
+  get alpnProtocol() {
+    return this[kAlpnProtocol];
+  }
+
+  get originSet() {
+    return this[kState].originSet ?
+      Array.from(this[kState].originSet) : undefined;
```

Comment:
This property's name implies a `Set` instance yet this returns an `Array`.

Reply:
The origin set terminology is adopted from the ORIGIN frame spec. It is a bit confusing. Do you have a suggested alternative?

Reply:
The name is great; I'm just wondering if it could return the `Set` (or clone thereof).

Reply:
"Origin Set" is the name within the spec, I don't think that should determine whether we use `Array` or `Set` as the data structure. I don't know if we have any APIs that currently return a Set

Reply:
Returning a `Set` may imply that it's mutable by the user, which is not the case. Yes, an Array is still mutable, but the property returns a new Array instance on every call. I'd prefer to keep the name and the array return.